src/golang/lib/job/databricks.go

```diff
@@ -0,0 +1,189 @@
+	databricksClient, err := databricks_lib.NewWorkspaceClient(
+		conf.WorkspaceUrl,
+		conf.AccessToken,
+	)
+	if err != nil {
+		return nil, errors.Wrap(err, "Unable to create Databricks Workspace Client.")
+	}
+
+	return &DatabricksJobManager{
+		databricksClient: databricksClient,
+		conf:             conf,
+		runMap:           map[string]int64{},
+	}, nil
+}
+
+func (j *DatabricksJobManager) mapJobTypeToFile(spec Spec) (string, string, error) {
+	if spec.Type() == FunctionJobType {
+		functionSpec, ok := spec.(*FunctionSpec)
+		if !ok {
+			return "", "", ErrInvalidJobSpec
+		}
+
+		functionSpec.FunctionExtractPath = defaultFunctionExtractPath
+		storageConfig, err := spec.GetStorageConfig()
+		if err != nil {
+			return "", "", errors.Wrap(err, "Spec unexpectedly has no storage config.")
+		}
+		storageConfig.S3Config.AWSAccessKeyID = j.conf.AwsAccessKeyId
+		storageConfig.S3Config.AWSSecretAccessKey = j.conf.AwsSecretAccessKey
+
+		specStr, err := EncodeSpec(spec, JsonSerializationType)
+		if err != nil {
+			return "", "", errors.Wrap(err, "Unable to encode spec.")
+		}
+		return databricksFunctionScript, specStr, nil
+
+	} else if spec.Type() == ParamJobType {
+		specStr, err := EncodeSpec(spec, JsonSerializationType)
+		if err != nil {
+			return "", "", errors.Wrap(err, "Unable to encode spec.")
+		}
+		return databricksParamScript, specStr, nil
+
+	} else if spec.Type() == AuthenticateJobType ||
+		spec.Type() == LoadJobType ||
+		spec.Type() == ExtractJobType ||
+		spec.Type() == LoadTableJobType ||
+		spec.Type() == DeleteSavedObjectsJobType ||
+		spec.Type() == DiscoverJobType {
+		specStr, err := EncodeSpec(spec, JsonSerializationType)
+		if err != nil {
+			return "", "", errors.Wrap(err, "Unable to encode spec.")
+		}
+		return databricksDataScript, specStr, nil
+
+	} else if spec.Type() == SystemMetricJobType {
+		specStr, err := EncodeSpec(spec, JsonSerializationType)
+		if err != nil {
+			return "", "", errors.Wrap(err, "Unable to encode spec.")
+		}
+		return databricksMetricScript, specStr, nil
+
+	} else {
+		return "", "", errors.New("Unsupported JobType was passed in.")
+	}
+}
+
+func (j *DatabricksJobManager) Config() Config {
+	return j.conf
+}
+
+func (j *DatabricksJobManager) Launch(
+	ctx context.Context,
+	name string,
+	spec Spec,
+) error {
+	log.Infof("Running %s job %s.", spec.Type(), name)
+
+	scriptFile, specStr, err := j.mapJobTypeToFile(spec)
+	if err != nil {
+		return err
+	}
+	storageConfig, err := spec.GetStorageConfig()
+	if err != nil {
+		return errors.Wrap(err, "Spec unexpectedly has no storage config.")
+	}
+	bucket := storageConfig.S3Config.Bucket
+	pythonFilePath := fmt.Sprintf("%s/%s", bucket, scriptFile)
+
+	jobId, err := databricks_lib.CreateJob(ctx, j.databricksClient, name, j.conf.S3InstanceProfileArn, pythonFilePath)
+	if err != nil {
+		return errors.Wrap(err, "Error creating job in Databricks.")
+	}
+	log.Info("SPEC:")
```

Comment:
nit: "DATABRICKS SPEC: " might be more informative in the logs.

Reply:
nit: We can combine the 2 log statements into: `log.Info("Databricks Spec: \n %v", specStr)`

Reply:
I also feel these logs can be moved to caller. Not required for this PR though